src/SimulationRunner.cc

```diff
@@ -567,10 +570,19 @@ void SimulationRunner::ProcessSystemQueue()
         this->postUpdateStartBarrier->Wait();
         if (this->postUpdateThreadsRunning)
         {
-          system->PostUpdate(this->currentInfo, this->entityCompMgr);
+          auto terminate = this->threadsToTerminate.find(parentEntity);
+          if (terminate != this->threadsToTerminate.end()) {
+            gzdbg << "Terminating thread " << id << ", " << parentEntity <<"\n";
+            this->postUpdateStartBarrier->Drop();
+            this->postUpdateStopBarrier->Drop();
+            break;
+          }
+          this_thread_system->PostUpdate(this->currentInfo, this->entityCompMgr);
```

Comment:
the linter is complaining that this line is over 80 characters

nice job finding the bug!